CsharpExtras/Map/Dictionary/Variant/Semi/IInKeyOutValueSemiDictionary.cs

```diff
@@ -0,0 +1,34 @@
+﻿using CsharpExtras.Map.Dictionary.Base;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace CsharpExtras.Map.Dictionary.Variant.Semi
+{
+    public interface IInKeyOutValueSemiDictionary<in K, out V> : IDictionaryBase,
+        IInKeySemiDictionaryBase<K>, IOutValueSemiDictionaryBase<V>
+    {
+        V this[K key] { get; }
+        ISuccessTuple<V> TryGetValue(K key);
+    }
+
+    public interface ISuccessTuple<out V>

```

Comment:
You might want to split this into its own file. Same for the implementation class below.